core/src/main/java/org/apache/calcite/rex/RexUtil.java

```diff
@@ -3191,4 +3231,75 @@ private static class SearchExpandingShuttle extends RexShuttle {
       }
     }
   }
+
+  /**
+   * Reorders some of the operands in this expression so structural comparison.
+   *
+   * <p>This is mostly the same as {@link SubstitutionVisitor#canonizeNode}, except
+   * that comparator is based on the {@link SqlKind}'s ordinal rather than string
+   * representation lexicographic order, which allows constant to always be on the
+   * right side of the expression. See {@link RexNormalize#reorderOperands} for details.
+   */
+  public static RexNode canonizeNode(RexBuilder rexBuilder, RexNode expression) {
```

Comment:
I think it preferable to have helper functions for interacting with the existing nodes then creating a copy.

Reply:
Ok. I'll fix this.